packages/components/Notification/src/Notification.tsx

```diff
@@ -30,15 +35,24 @@ export const Notification = compose<NotificationType>({
   },
   useRender: (userProps: NotificationProps, useSlots: UseSlots<NotificationType>) => {
     const Slots = useSlots(userProps, (layer) => notificationLookup(layer, userProps));
+    const isBar = ['primaryOutlineBar', 'primaryBar', 'neutralBar'].includes(userProps.variant);
+    const rootStyle = useMemo(() => {
+      const marginHorizontal = isBar ? 0 : 16;
+      return { variant: userProps.variant, marginHorizontal: marginHorizontal };
+    }, ['variant', 'isBar']);
```

Comment:
`variant` is not a styeProp. The final object returned is a mix of one Notification prop, and one style prop. We're also not modifying variant at all, so I'd rather it not be in the returned object. This code as written would not place "marginHorizontal" inside Slots.root's style I believe. 

To enforce type checking so that VSCode's intelligence can check this error, we can add some type annotations:

```suggestion
    const rootStyle: ViewStyle = useMemo(() => {
      const marginHorizontal = isBar ? 0 : 16;
      return { marginHorizontal: marginHorizontal };
    }, [ 'isBar']);
```

Reply:
resolved offline